package.json

```diff
@@ -24,6 +25,7 @@
     "lib": "./lib"
   },
   "dependencies": {
+    "@types/q": "^1.5.1",
```

Comment:
Need move to `devDependencies`

Reply:
Are you sure? That must be provided with coa dependency by default. An absence of this package leads to TS errors while the consumer's code compilation.

Reply:
Ofcourse it has no sense in pure javascript projects. But crucial in typescript projects.

Reply:
Hello! I wasted some time seeking best practices about such dependencies. I met this [answer](https://stackoverflow.com/questions/45176661/how-do-i-decide-whether-types-goes-into-dependencies-or-devdependencies) at StackOverflow and this [article](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) at the TS documentation.

In a nutshell, if your typing reference some dependency lib then you should include this lib into 'dependencies' section, not in 'devDependencies'.

There are two methods return Q.Promise, that's why we need `@types/q` among the dependencies.

As an alternative solution, we could replace Q.Promise from these lines [coa.d.ts:42-43](https://github.com/alexbadm/coa/blob/master/coa.d.ts#L42-L43) by `any` or `Promise` which would be rough.